### PR TITLE
Stop having two types of version.

### PR DESCRIPTION
**Pull request by jgiszczak**
Will get rid of support for two types in appbase later.

### CODE REVIEW

File: programs/eosd/main.cpp [hunk 1]
```diff
@@ -27,9 +27,8 @@ using namespace eosio;
 int main(int argc, char** argv)
 {
    try {
-      app().set_version(eosio::eosd::config::version_str);
-      app().set_version(eosio::eosd::config::version_int);
-      ilog("eosd version ${ver}", ("ver", app().version()));
+      app().set_version(eosio::eosd::config::version);
+      ilog("eosd version ${ver}", ("ver", eosio::eosd::config::itoh((uint32_t)app().version_int())));
```
Comment (heifner) [inline programs/eosd/main.cpp]: I would prefer a `static_cast<uint32_t>()`

Comment (jgiszczak) [reply programs/eosd/main.cpp]: I'm converting it to hex in order to provide an easily human readable match between the reported version from a running daemon and git's hash.  Forcing people running the daemon to manually convert from decimal to hex to do that check is very unfriendly.

Comment (heifner) [reply programs/eosd/main.cpp]: Right. I was just saying this:
`eosio::eosd::config::itoh((uint32_t)app().version_int())`
to
`eosio::eosd::config::itoh(static_cast<uint32_t>(app().version_int()))`

Comment (jgiszczak) [reply programs/eosd/main.cpp]: Oh.  Sure, will do.

**Pull request merged**